spec/controllers/admin/users_controller_spec.rb

```diff
@@ -3,164 +3,132 @@
 require 'rails_helper'
 
 RSpec.describe Admin::UsersController, type: :request do
-  context 'when JWT Token is valid' do
-    let!(:current_user) { create(:user) }
-    let!(:another_user) { create(:user, name: 'Second user', slug: 'second-user-1', email: 'second.user@mail.com') }
-
-    let(:jwt_token) do
-      request_params = {
-        email: current_user.email,
-        password: current_user.password
-      }
+  let!(:current_user) { create(:user, role: :admin) }
+  let!(:another_user) { create(:user, name: 'Second user', slug: 'second-user-1', email: 'second.user@mail.com') }
+
+  let(:jwt_token) { JsonWebToken.encode(user_id: current_user.id) }
+
+  let(:json) { JSON.parse(response.body) }
 
-      post(auth_login_path, params: request_params)
+  describe '#index' do
+    subject { get(admin_users_path, headers: { Authorization: jwt_token }) }
 
-      JSON.parse(response.body)['token']
+    it 'returns all users' do
+      subject
+
+      expect(response).to have_http_status :ok
+      expect(json[0]['email']).to eq(current_user.email)
+      expect(json[1]['email']).to eq(another_user.email)
     end
 
-    let(:json) { JSON.parse(response.body) }
+    include_examples 'Invalid JWT Token'
+    include_examples 'Unauthorize normal user'
```

Comment:
Eu gostaria mais de:
```suggestion
    include_examples 'Unauthorize user that does not have admin role'
```
As vezes eu posso tentar entender o que que tá acontecendo no seu código pelos testes... E se eu ver "normal user" eu vou pensar "mas o que que é um normal user???" e se vc coloca que é "user that does not have admin role" eu já consigo entender melhor o que que tá acontecendo...